app/dcr_monitor/imagebuilder/kaniko.go

```diff
@@ -0,0 +1,149 @@
+package imagebuilder
+
+import (
+	"bufio"
+	"context"
+	"io"
+	"os"
+	"regexp"
+
+	"github.com/cloudwego/hertz/pkg/common/hlog"
+	"github.com/pkg/errors"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+type ImageInfo struct {
+	Image  string
+	Digest string
+}
+
+type ImageBuilder interface {
+	CheckImageBuilderStatusAndGetInfo(string) (bool, *ImageInfo, error)
+}
+
+type KanikoImageBuilder struct {
+	ctx       context.Context
+	clientSet *kubernetes.Clientset
+	namespace string
+}
+
+func NewKanikoImageBuilder() (*KanikoImageBuilder, error) {
+	var err error
+	clusterConfig, err := rest.InClusterConfig()
+	if err != nil {
+		panic(err)
+	}
+
+	clientSet, err := kubernetes.NewForConfig(clusterConfig)
+	if err != nil {
+		panic(err)
+	}
+
+	RunningNameSpaceByte, err := os.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/namespace")
+	if err != nil {
+		panic(err)
+	}
+	namespace := string(RunningNameSpaceByte)
+
+	return &KanikoImageBuilder{
+		ctx:       context.Background(),
+		clientSet: clientSet,
+		namespace: namespace,
+	}, nil
+}
+
+func (b *KanikoImageBuilder) CheckImageBuilderStatusAndGetInfo(uuid string) (bool, *ImageInfo, error) {
+
+	k8sJobName := "kaniko-" + uuid
+	k8sJob, err := b.clientSet.BatchV1().Jobs(b.namespace).Get(b.ctx, k8sJobName, metav1.GetOptions{})
+	if err != nil {
+		hlog.Errorf("[KanikoJobMonitor]failed to get job: %v", err)
+		return false, nil, errors.Wrap(err, "failed to get job")
+	}
+
+	if len(k8sJob.Status.Conditions) == 0 {
+		hlog.Infof("[KanikoJobMonitor]job %v is still running", k8sJob.Name)
+		return false, nil, nil
+	}
+
+	hlog.Infof("[KanikoJobMonitor]job name: %v, job status: %v", k8sJob.Name, k8sJob.Status.Conditions[0].Type)
+
+	if k8sJob.Status.Conditions[0].Type == batchv1.JobComplete {
+		image, digest, err := b.getImageDigestAndDeleteJob(k8sJob.Name)
```

Comment:
How about devide the `getImageDigestAndDeleteJob` function into two functions. Something like `getImageDigest` and `deleteJob`

Reply:
done